src/routes/session/session-get.test.ts

```diff
@@ -0,0 +1,93 @@
+import supertest from "supertest";
+import express, { Express } from "express";
+import { getFrontendApp } from "../../app";
+import { getLogger } from "config/logger";
+
+describe("Session GET", () => {
+	let app: Express;
+
+	beforeEach(() => {
+		app = express();
+		app.use((request, _, next) => {
+			request.log = getLogger("test");
+			next();
+		});
+	});
+
+	describe("Frontend", () => {
+		beforeEach(() => {
+			app.use(getFrontendApp({
+				getSignedJsonWebToken: () => "",
+				getInstallationAccessToken: async () => ""
+			}));
+		});
+
+		describe("Testing session", () => {
+			it("should return session.hbs", () =>
+				supertest(app)
+					.get("/session/jira/atlassian-connect.json")
+					.expect(200)
+					.then(response => {
+						expect(response.text.includes("Logging you into GitHub")).toBeTruthy();
+						expect(response.text.includes("Redirecting to your GitHub Enterprise Server instance")).toBeFalsy();
+						expect(response.text.includes("Redirecting to your GitHub Cloud instance")).toBeFalsy();
+					}));
+
+			it("should return error asking for redirect url", () =>
+				supertest(app)
+					.get("/session")
+					.expect(500)
+					.then(response => {
+						expect(response.text.includes("Missing redirect url for session")).toBeTruthy();
+						expect(response.text.includes("Logging you into GitHub")).toBeFalsy();
+						expect(response.text.includes("Redirecting to your GitHub Enterprise Server instance")).toBeFalsy();
+						expect(response.text.includes("Redirecting to your GitHub Cloud instance")).toBeFalsy();
+					}));
+		});
+
+		describe("Testing session for GHE", () => {
+			it("should return ghe-loading.hbs", () =>
+				supertest(app)
+					.get("/session/enterprise/jira/atlassian-connect.json")
+					.expect(200)
+					.then(response => {
+						expect(response.text.includes("Logging you into GitHub")).toBeFalsy();
+						expect(response.text.includes("Redirecting to your GitHub Enterprise Server instance")).toBeTruthy();
+						expect(response.text.includes("Redirecting to your GitHub Cloud instance")).toBeFalsy();
+					}));
+
+			it("should return error asking for redirect url", () =>
+				supertest(app)
+					.get("/session/enterprise")
+					.expect(500)
```

Comment:
/nit 
I think 4xx will state the problem better in this case, as is the client's error that forgot to supply a redirect url.

Reply:
Status updated to `400`